File: java-sample/src/main/java/io/github/simonscholz/Base64Main.java
package io.github.simonscholz;

import io.github.simonscholz.qrcode.QrCodeApi;
import io.github.simonscholz.qrcode.QrCodeConfig;
import io.github.simonscholz.qrcode.QrCodeConfig.Builder;
import io.github.simonscholz.qrcode.QrCodeExtensionsKt;
import io.github.simonscholz.qrcode.QrCodeFactory;

import java.awt.image.BufferedImage;

public class Base64Main {

    public static void main(final String[] args) {
        final QrCodeApi qrCodeApi = QrCodeFactory.createQrCodeApi();
        final QrCodeConfig qrCodeConfig = new Builder("https://simonscholz.github.io/").build();
        // using createBase64QrCodeImage
        final String base64Image = qrCodeApi.createBase64QrCodeImage(qrCodeConfig);
        System.out.println(base64Image);

        // using createBase64QrCodeImageWithSize
        final BufferedImage qrCodeImage = qrCodeApi.createQrCodeImage(qrCodeConfig);
        final String base64Image2 = QrCodeExtensionsKt.toBase64(qrCodeImage);
        System.out.println(base64Image2);
    }
}

File: kotlin-sample/src/main/kotlin/io/github/simonscholz/Base64Main.kt
package io.github.simonscholz

import io.github.simonscholz.qrcode.QrCodeConfig
import io.github.simonscholz.qrcode.QrCodeFactory
import io.github.simonscholz.qrcode.toBase64

/**
 * This is a sample to demonstrate the usage of the qr-code library with a base64 encoded image.
 *
 * It creates a qr code as a base64 encoded image and prints it to the console.
 */
fun main() {
    val createQrCodeApi = QrCodeFactory.createQrCodeApi()
    QrCodeConfig.Builder("https://simonscholz.dev/").build().run {
        val base64QrCodeImage = createQrCodeApi.createBase64QrCodeImage(this)
        println(base64QrCodeImage)

        val toBase64 = createQrCodeApi.createQrCodeImage(this).toBase64()
        println(toBase64)
    }
}
